src/models/metrics.py

```diff
@@ -0,0 +1,137 @@
+"""Functions for calculating metrics from a GeoGraph."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Optional
+
+import networkx as nx
+import numpy as np
+
+if TYPE_CHECKING:
+    from src.models import geograph
```

Comment:
Nice, didn't know there is a variable like this. Should start to use this as well!